src/Configuration/src/ConfigServerBase/ConfigServerHostedService.cs

```diff
@@ -30,7 +31,17 @@ public ConfigServerHostedService(IConfigurationRoot configuration, ILoggerFactor
             throw new ArgumentNullException(nameof(configuration));
         }
 
-        _configuration = configuration.Providers.First(provider => provider is ConfigServerConfigurationProvider) as ConfigServerConfigurationProvider;
+        if (configuration.Providers.Count() == 1 && configuration.Providers.First() is PlaceholderResolverProvider resolverProvider)
+        {
+            _configuration =
+                resolverProvider.Providers.First(provider => provider is ConfigServerConfigurationProvider) as
+                    ConfigServerConfigurationProvider;
+        }
+        else
+        {
+            _configuration = configuration.Providers.First(provider => provider is ConfigServerConfigurationProvider) as ConfigServerConfigurationProvider;
```

Comment:
Application of @bart-vmware suggestion
```suggestion
            _configuration = resolverProvider.Providers.OfType<ConfigServerConfigurationProvider>().First();
        }
        else
        {
            _configuration = configuration.Providers.OfType<ConfigServerConfigurationProvider>().First();
```

Reply:
![image](https://user-images.githubusercontent.com/3947063/178749486-fca021c8-7a4e-42e5-a607-95d831d58154.png)